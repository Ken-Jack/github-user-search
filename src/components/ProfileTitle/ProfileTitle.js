// @flow

import React from 'react';
import {connect} from 'react-fela';

type Props = {
  userLink: string,
  username: string,
  name: string | null,
  styles: Object,
};

function ProfileTitle(props: Props) {
  const {
    name,
    username,
    userLink,
    styles,
  } = props;

  return (
    <div className={styles.ProfileTitle_root}>
      {name
        ? <h1 className={styles.ProfileTitle_name}>{name}</h1>
        : null
      }
      <p className={styles.ProfileTitle_username}>
        <a href={userLink}>@{username}</a>
      </p>
    </div>
  );
}

const styles = {
  ProfileTitle_root: () => ({
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'baseline',
  }),

  ProfileTitle_username: () => ({
    color: '#555',
  }),

  ProfileTitle_name: () => ({
    marginRight: '5px',
    fontSize: '28px',
  }),
};

export default connect(styles)(ProfileTitle);
