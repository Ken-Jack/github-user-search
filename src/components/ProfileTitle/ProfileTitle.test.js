import React from 'react';
import felaSnapshot from 'test-util/fela-snapshot';
import ProfileTitle from './ProfileTitle';

describe('Component: ProfileTitle', () => {

  it('should render', () => {
    const component = felaSnapshot(
      <ProfileTitle
        name="Simon"
        username="simonsmith"
        userLink="http://simonsmith.io"
      />
    );
    expect(component).toMatchSnapshot();
  });

});
