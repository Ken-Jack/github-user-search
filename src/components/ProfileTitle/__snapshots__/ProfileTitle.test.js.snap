// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProfileTitle should render 1`] = `
Object {
  "component": <div
    className="a b c"
  >
    <h1
      className="e f"
    >
      Simon
    </h1>
    <p
      className="d"
    >
      <a
        href="http://simonsmith.io"
      >
        @
        simonsmith
      </a>
    </p>
  </div>,
  "styles": "
    .a{display:flex}
    .b{flex-wrap:wrap}
    .c{align-items:baseline}
    .d{color:#555}
    .e{margin-right:5px}
    .f{font-size:28px}",
}
`;
